tests/unit/amo/test_withExperiment.js

```diff
@@ -304,6 +309,18 @@ describe(__filename, () => {
     expect(root).toHaveProp('isExperimentEnabled', false);
   });
 
+  it('throws an exception for a badly formatted experimentId', () => {
+    expect(() => {
+      render({ experimentProps: { id: 'bad-id' } });
+    }).toThrow(/id must match the pattern YYYYMMDD_experiment_id/);
+  });
+
+  it('does not have any invalid experiment ids defined in the config', () => {
+    for (const experimentId of Object.keys(config.get('experiments'))) {
```

Comment:
Maybe add a comment to explain that when this fails, it's because the id is badly formatted and we expect a date, etc. (or to go read the comment in `config/default.js`).